### ISSUE TITLE
"No such table exception" in functional tests with multiple requests and SQLite memory database

### ISSUE THREAD

**Issue by mi-schi** [symfony/symfony#16252]
Hello

to speed up the functional tests I use a SQLite memory database. Here is the config:
```
doctrine:
  dbal:
    default_connection: default
    connections:
      default:
        driver:  pdo_sqlite
        memory: true
```
In the ```setUp``` method I create the tables:
```
$cmf = $em->getMetadataFactory();
$classes = $cmf->getAllMetadata();

$schemaTool = new SchemaTool($em);
$schemaTool->createSchema($classes);
```

With one request in one test method, everything is fine:
```
public function testDoSomeRequest()
{
    $client = self::createClient();
    $client->request('GET', '/nice-url');

    $this->assertTrue($client->getResponse()->isRedirect());
}
```
But if I want to send multiple requests in one test the database tables has gone away.
```
public function testDoSomeRequest()
{
    $client = self::createClient();
    $client->request('GET', '/nice-url');
    $form = $crawler->selectButton('_submit')->form();

    $client->submit($form, [
        '_username' => 'username',
        '_password' => 'password'
    ]);

    $this->assertTrue($client->getResponse()->isRedirect());
}
```

The ```submit``` method is a wrapper for the ```request```. This method calls the ```doRequest``` method. In this method the [Symfony\Bundle\FrameworkBundle\Client](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bundle/FrameworkBundle/Client.php) shutdown the kernel (line 119). This triggers the ```Doctrine\Bundle\DoctrineBundle\DoctrineBundle::shutdown``` method. There the connections were closed (line 131). The tables are lost. 

**Fast solution:**
I write a wrapper for the client. The ```doRequest``` method contains only the content of the parent [Symfony\Component\HttpKernel\Client](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/HttpKernel/Client.php) .
```
/**
 * @return Client
 */
protected function createClient()
{
    $client = $this->getContainer()->get('test.client');

    return new Client($client->getKernel(), [], $client->getHistory(), $client->getCookieJar());
}
```
```
<?php

namespace AppBundle\Tests\Functional;

use Symfony\Bundle\FrameworkBundle\Client as BaseClient;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\HttpKernel\TerminableInterface;

/**
 * Class Client
 *
 * @package AppBundle\Tests\Functional
 */
class Client extends BaseClient
{
    /**
     * @param Request $request A Request instance
     *
     * @return Response A Response instance
     */
    protected function doRequest($request)
    {
        $response = $this->kernel->handle($request);

        if ($this->kernel instanceof TerminableInterface) {
            $this->kernel->terminate($request, $response);
        }

        return $response;
    }
}
```

Is there a better way to send multiple requests in one test with a memory database? Why should a "hasPerformedRequest" (the second and further) request shutdown the kernel?

**Comment by Tobion** [comment]
Did you try to use `disableReboot()`? It looks like that what your trying to achieve.

**Comment by mi-schi** [comment]
`disableReboot()` is not available in symfony 2.3. Time to update to version 2.7 ;)
